### PR TITLE
Fixed #5586 Threading Implementation Typos

### LINKED ISSUE #5586: Incorrect Directory Reference to numba/npyunfunc
## Reporting a bug

- [x] I am using the latest released version of Numba (most recent is visible in
 the change log (https://github.com/numba/numba/blob/master/CHANGE_LOG).
- [x] I have included below a minimal working reproducer (if you are unsure how
 to write one see http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports).

### PR DESCRIPTION
**Pull request by seanlaw**
This fix addresses typos in a document and does not conflict with any code.

### CODE REVIEW

File: docs/source/developer/threading_implementation.rst [hunk 1]
```diff
@@ -29,14 +29,14 @@ The relevant source files referenced in this document are
...
-- ``numba/npyufunc/parallel.py``
+- ``numba/np/ufunc/parallel.py``
 
   This file contains the Python and JIT compatible wrappers for
   ``set_num_threads()``, ``get_num_threads()``, and ``get_thread_id()``, as
   well as the code that loads the above libraries into Python and launches the
   threadpool.
 
-- ``numba/npyufunc/parfor.py``
+- ``numba/np/ufunc/parfor.py``
```
Comment (stuartarchibald) [inline docs/source/developer/threading_implementation.rst:39]: This is now `numba/parfors/parfor_lowering.py`

File: docs/source/developer/threading_implementation.rst [hunk 2]
```diff
@@ -47,7 +47,7 @@ Thread masking
 --------------
 
 As part of its design, Numba never launches new threads beyond the threads
-that are launched initially with ``numba.npyufunc.parallel._launch_threads()``
+that are launched initially with ``numba.np/ufunc.parallel._launch_threads()``
```
Comment (stuartarchibald) [inline docs/source/developer/threading_implementation.rst:50]:
```suggestion
that are launched initially with ``numba.np.ufunc.parallel._launch_threads()``
```
refers to module not path?!

File: docs/source/developer/threading_implementation.rst [hunk 3]
```diff
@@ -212,7 +212,7 @@ The general pattern for using ``get_num_threads`` in code generation is
 
    # Pass num_threads through to the appropriate backend function here
 
-See the code in ``numba/npyufunc/parfor.py``.
+See the code in ``numba/np/ufunc/parfor.py``.
```
Comment (stuartarchibald) [inline docs/source/developer/threading_implementation.rst:215]: This is now `numba/parfors/parfor_lowering.py`

Comment (stuartarchibald) [reply docs/source/developer/threading_implementation.rst:215]: Think this still needs an update?

File: docs/source/developer/threading_implementation.rst (hunk 1 above)
Comment (seanlaw) [reply docs/source/developer/threading_implementation.rst]: done

File: docs/source/developer/threading_implementation.rst (hunk 2 above)
Comment (seanlaw) [reply docs/source/developer/threading_implementation.rst]: You're right. I've reverted this back

File: docs/source/developer/threading_implementation.rst (hunk 3 above)
Comment (seanlaw) [reply docs/source/developer/threading_implementation.rst]: I missed this one. Thanks!

**Comment by seanlaw** [comment]
> Thanks for the patch and fixes!

And thank you and the team for all of your work on Numba!

**Pull request merged**